kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/AssetConstants.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2017 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ *******************************************************************************/
+package org.eclipse.kura.asset.provider;
+
+import org.eclipse.kura.type.DataType;
+
+/**
```

Comment:
`AssetConstants` is an integral part of the current implementation and that's why I kept it in API to make it available for all the consumers. The same idea is followed by OSGi Alliance and that's why `Constants.java` has been kept in API Bundle in `org.osgi.framework` package. In addition, extracting `AssetConstants` resulted in code duplication in multiple bundles as well. I personally think that the documentation of configuring asset channels also is not suitable in `AssetConstants`. Due to release constraint, don't put the documentation here. You can keep this documentation in `BaseAsset` and in web bundle, use this `AssetConstants` too as web bundle also imports `org.eclipse.kura.asset.provider`

The ideal way is to extract `AssetConfiguration` as an interface and keep the default implementation in `Asset Provider` and the implementation will have this javadoc. 

To wrap up, have the documentation in `BaseAsset` and keep the AssetConstants in the exported package of `Asset Provider`. Use the exported `AssetConstants` in Web bundle too. Don't duplicate the constants in web bundle again.